### DISCUSSION TITLE
Windows build broken - Duplicate targets for freetype/lcms/yaml_cpp

### DISCUSSION THREAD

**Discussion by richardssam** [AcademySoftwareFoundation/OpenRV#298] [Build Issues]
Doing a fresh checkout of the repo on windows is failing to build. I'm seeing errors:
`CMake Error at src/pub/freetype/CMakeLists.txt:62 (ADD_LIBRARY):
  ADD_LIBRARY cannot create target "freetype" because another target with the
  same name already exists.  The existing target is a static library created
  in source directory "C:/OpenRV.new/src/pub/freetype".  See documentation
  for policy CMP0002 for more details.

CMake Error at src/pub/lcms/CMakeLists.txt:41 (ADD_LIBRARY):
  ADD_LIBRARY cannot create target "lcms" because another target with the
  same name already exists.  The existing target is a shared library created
  in source directory "C:/OpenRV.new/src/pub/lcms".  See documentation for
  policy CMP0002 for more details.

CMake Error at src/pub/yaml_cpp/CMakeLists.txt:47 (ADD_LIBRARY):
  ADD_LIBRARY cannot create target "yaml_cpp" because an imported target with
  the same name already exists.

CMake Error at src/pub/yaml_cpp/CMakeLists.txt:52 (TARGET_INCLUDE_DIRECTORIES):
  TARGET_INCLUDE_DIRECTORIES may only set INTERFACE properties on IMPORTED
  targets

CMake Error at src/pub/yaml_cpp/CMakeLists.txt:58 (TARGET_COMPILE_OPTIONS):
  TARGET_COMPILE_OPTIONS may only set INTERFACE properties on IMPORTED
  targets

`

Which I believe is caused by changes to https://github.com/AcademySoftwareFoundation/OpenRV/commits/main/cmake/dependencies/CMakeLists.txt from commit #270 since we now have the freetype subdirectory added twice.

**Comment by bernie-laberge** [comment]
Hello, I apologize for the delay ! 
Let me do a fresh windows build from scratch with the latest Open RV and I'll get back to you.

**Reply by bernie-laberge** [reply]
OK so I just did a complete windows build from scratch of the latest OpenRV on my Windows 11 machine and everything built successfully. A typical case of "it works on my machine". So it didn't help me identify the root cause of your build failure which might be experienced by others I am sure.

Now about the freetype target: it should only be added once.
ADD_SUBDIRECTORY(../../src/pub/freetype src/pub/freetype) # required by oiio

It is not added twice (at least it shouldn't :-) ) 
It is used here to specify a different binary directory than the source directory:
ADD_SUBDIRECTORY(source_dir [binary_dir] [EXCLUDE_FROM_ALL] [SYSTEM])

I suspect that some remnants from whatever was before commit 270 is causing this issue...
I would suggest to do a fresh clone from OpenRV and build everything from scratch to see if this error goes away.
If you still end up with the same error then it would be great to see where this redundant freetype target is coming from.

**Reply by richardssam** [reply]
I thought it was a fresh checkout, but I'll try again. Thanks so much for testing.
